libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -884,10 +884,6 @@ float AP_GPS::get_lag(uint8_t instance) const
     if (_delay_ms[instance] > 0) {
         // if the user has specified a non zero time delay, always return that value
         return 0.001f * (float)_delay_ms[instance];
-    } else if (drivers[instance] == nullptr || state[instance].status == NO_GPS) {
-        // the user has not specified a value and we cannot determine it from the GPS type
-        // so return a default delay of 1 measurement interval
-        return 0.001f * (float)_rate_ms[instance];
```

Comment:
@priseborough This is actually wrong, we should check for a non-existent driver in the requested instance (it would be a wrong use of the function from the caller but we should prevent it here). I think we should return a default 200 ms in that case.
Would you prefer to use the _rate_ms in case a backend exists but doesn't have it's own defined lag (so every GPS except Ublox)? I can do that if you want.